articles/choosing-between-firebase-and-mongodb/index.md

```diff
@@ -0,0 +1,95 @@
+4. Firebase Storage
+5. Firebase Analytics
+### Advantages of Firebase
+1. It is free.
+Developers can create as more than one project in Firebase for free. Developers don't need to buy a premium license to use Firebase.
+2. Concise Documentation
+Firebase SDK is documented and has plenty of examples over its website.
+3. Improved Database
+Firebase is an improved database since it provides better performance by storing large chunks of data in JSON format.
+4. It is fast.
+Applications and websites developed using Firebase are faster because its cloud services lead to faster fetching and displaying of data on the screens.
+### Disadvantages of using Firebase
+1. Hosted solution
+Firebase is a hosted solution so the user doesn't have any root access to the location where the data is stored.
+2. Costs
+Firebase free plan limits up to 50 connections and 1,000 Megabytes of storage.
+3. Data Migration Problem.
+Firebase stores its data in JSON format so managing large amounts of structured data need the developer to write powerful queries in SQL.
+4. Inconvenient Data Storage
+Firebase cannot handle complex query structures. It limits the developer to combine query options. Because of this limitation, one needs to arrange the documents by date or filter them using a search query.
+5. No staging environment.
+Firebase does not allow the developer to test the database in a staging environment.
+### Differences betweeen MongoDB and Firebase
+According to [GeeksforGeeks](https://www.geeksforgeeks.org/difference-between-firebase-and-mongodb/), the main differences between MongoDB  and Firebase are:
+|MongoDB|Firebase|
+|---|---|
+|MongoDB Inc developed MongoDB.| Google developed Firebase|
+|MongoDB supports C, C#, Java, JavaScript, PHP, Lau, Python, R, Ruby as programming languages. | Firebase supports  Objective C, Java and JavaScript as programming languages.|
+|MongoDB is a Platform-as-a-Service solution.| Firebase is a Backend-as-a-Service solution.|
+|MongoDB is suitable for large-scale applications.| Firebase is suitable for small-scale applications.|
+|MongoDB is an open-source database. | Firebase is an open-source database.|
+|Proprietary protocol using JSON are used as APIs and other access methods.|Android, iOS, JavaScript API, RESTful HTTP API are used as APIs and other access methods.|
+|Server operating systems for MongoDB are Solaris, Linux, OS X, Windows. | Server operating systems for Firebase are hosted.|
+### Conclusion
+Every database has its unique features, and it is designed to provide solutions for different problems. The choice of each database depends on its benefits and drawbacks, and also the business requirements. Both MongoDB and Firebase have their pros and cons. After all the advantages, and disadvantages, comparisons and the description, the developer will have to choose which database they can integrate into their application or website.
```

Comment:
Generic. This should be a recap of the specific similarities and differences you discuss, as a summary.